src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -16,10 +18,10 @@ public class TagHelperExecutionContext
     {
         private readonly List<ITagHelper> _tagHelpers;
         private readonly Action<HtmlEncoder> _startTagHelperWritingScope;
-        private readonly Func<TagHelperContent> _endTagHelperWritingScope;
-        private TagHelperContent _childContent;
+        private readonly Func<IHtmlContent> _endTagHelperWritingScope;
+        private IHtmlContent _childContent;
         private Func<Task> _executeChildContentAsync;
-        private Dictionary<HtmlEncoder, TagHelperContent> _perEncoderChildContent;
+        private Dictionary<HtmlEncoder, IHtmlContent> _perEncoderChildContent;
```

Comment:
Why? None of these types should need to change and all of these consistently are or return `TagHelperContent` for realz.